docs/ADR/0010_qfpolkavm_pallet_execute.md

```diff
@@ -0,0 +1,68 @@
+## Title: Design of Polka VM pallet's execute() extrinsic
+
+## Date
+2025-05-20
+Last status update: 2025-05-21
+
+## Status
+- [X] Proposed
+- [X] Accepted
+- [ ] Rejected
+- [ ] Deprecated
+- [ ] Superseded by ADR #X
+
+### Implementation Status
+- [ ] Planned
+- [X] In Development
+- [ ] Implemented
+- [ ] Verified
+- [ ] Discontinued
+
+## People
+- **Decision-makers**: KP, Vas
+- **Consulted**: Andrew, Alisher
+- **Informed**:
+  - Svat
+
+## Context
+For running a smart contract we have to have implemented pallet's extrinsic execute() and at least one method inside the contract to run it (i.e. main()).
+
+## Problem
+We have introduced [pallet's execute method](https://github.com/QuantumFusion-network/qf-solochain/blob/main/pallets/qf-polkavm/src/lib.rs#L333), it has signature:
+```
+execute(origin, contract_address, to, value, user_data, gas_limit, gas_price)
+```
+, where:
+- `origin` is a caller address,
+- `contract_address` is an address of stored on-chain contract,
+- `user_data` is [SCALE] encoded structures passed to the contract's main() function,
+- `gas_limit` is a limit of the gas for executing of the contract behind contract_address,
+- `gas_price` is a price of the gas for auction; this argument is not implemented, it's for the future.
+
+As we can see the signature looks at least strage and contains legacy (from PoC) arguments, which are `to`, `value`.
+
+## Decision
+### Change signature of pallet's extrinsic to:
+```
+execute(origin, contract_address, data, gas_limit, storage_deposit_limit, gas_price)
+```
+, where:
+- `origin` is a caller address,
+- `contract_address` is an address of stored on-chain contract,
+- `data` is [SCALE] encoded structures passed to the contract's context and available via host function call,
```

Comment:
If serialization is up to the smart contract developer we should not specify SCALE as the requirement here.

Reply:
Correct comment, fixed in: d8a3ab7c894b9d6b073d63fb03c85f9fcd6dea82